eoxserver/render/mapserver/factories.py

```diff
@@ -669,6 +669,19 @@ def _create_raster_layer_objs(map_obj, extent, sr, data, filename_generator, env
     layer_obj.type = ms.MS_LAYER_RASTER
     layer_obj.status = ms.MS_ON
 
+    # if attempting to load NETCDF file, need to use a temporary file to extract only
+    # the required band data
+    if "NETCDF" in data:
+        # determine if attempting to load a specific band index
+        if ("https://" in data and data.count(":") == 4) or (not "https://" in data and data.count(":") == 3):
+            splitpath = data.rsplit(":",1)
+            data = splitpath[0]
+            index = int(splitpath[1]) + 1
+        temp_path = "/vsimem/%s" % uuid4().hex
+        # extract only desired index to new temporary file
+        gdal.Translate(temp_path, data, bandList=[index])
+        data = temp_path
```

Comment:
This `vsimem` path will work, however there is no automatic cleanup involved, this it will fill up the memory over time. There is a mechanism with the `filename_generator`, that will generate names for temporary files, which will subsequently be deleted. Look for example here: https://github.com/EOxServer/eoxserver/blob/e16ac44c8aeccb1ceb622be119ac84fc6a8dcf93/eoxserver/render/mapserver/factories.py#L711

Reply:
Thanks for your suggestion @constantinius, I have updated the code as you have suggested.